lib/bolt/cli.rb

```diff
@@ -441,12 +433,26 @@ def execute(options)
       raise e
     end
 
+    def with_bolt_executor(executor, &block)
+      Puppet.override(bolt_executor: executor, &block)
```

Comment:
Can we bind this block to the executor in bolt instead of having puppet aware of it?

Reply:
@adreyer , I think it's desirable to be able to use this block from Puppet too. That would enable a run_plan to report progress from the tasks that it run and to have full control over the blocks. We have already discussed moving `Target` to Bolt. The block would be "puppet safe" if called with a `Target` instance and a `Hash[String,Data]` instead of the current Ruby `node` and a hash using symbols.

Reply:
That makes sense. The format is basically what the transport proposal suggests.

Reply:
Could we move both `Target` & `ExecutionResult` to bolt then? The would allow for a clean implementation without performance penalty.

Reply:
Part of that work is in progress. I'm working on a prerequisite. See [PUP-8195](https://tickets.puppetlabs.com/browse/PUP-8195).

Reply:
👍
So can I just create & use plain old ruby classes or do I need to create something more specific?